Formula/nginx-unit.rb

```diff
@@ -0,0 +1,256 @@
+class NginxUnit < Formula
+  desc "Dynamic web and application server for multiple languages"
+  homepage "https://unit.nginx.org/"
+  url "https://unit.nginx.org/download/unit-1.1.tar.gz"
+  sha256 "7c66365f5ea87e7e8903bac9d639676947fe2ab7aa799ce35defc426b3409ee0"
+
+  depends_on "openssl"
+  depends_on "pcre"
+
+  def install
+    openssl = Formula["openssl"]
+    pcre = Formula["pcre"]
+
+    system "./configure",
+      "--prefix=#{prefix}",
+      "--bindir=#{bin}",
+      "--sbindir=#{bin}",
+      # This tells the unit daemon where it can load modules from. Configuring
+      # the location at runtime is painful. We want this to be in the
+      # $HOMEBREW_PREFIX so that other formulae can link in additional modules.
+      "--modules=#{HOMEBREW_PREFIX}/lib/unit",
+      "--state=#{var}/unit",
+      "--log=#{var}/log/unit.log",
+      "--pid=#{var}/run/unit.pid",
+      "--control=unix:#{var}/run/unit.control.sock",
+      "--cc-opt=-I#{pcre.opt_include} -I#{openssl.opt_include}",
```

Comment:
There aren't configure options for openssl and pcre specifically?

Reply:
No. Their [configure scripts](https://github.com/nginx/unit/blob/master/configure) are completely custom:

```
unit-1.1$ ./configure --help

./configure options:

  --cc=FILE            set C compiler filename, default: "clang"
  --cc-opt=OPTIONS     set additional C compiler options
  --ld-opt=OPTIONS     set additional linker options

  --prefix=DIRECTORY   set prefix for relative pathnames, default: none
  --bindir=DIRECTORY   set user executables directory name
                       default: "bin"
  --sbindir=DIRECTORY  set system admin executables directory name
                       default: "sbin"
  --modules=DIRECTORY  set modules directory name, default: "build"
  --state=DIRECTORY    set state directory name, default: "build"

  --pid=FILE           set pid filename, default: "unit.pid"
  --log=FILE           set log filename, default: "unit.log"

  --control=ADDRESS    set address of control API socket
                       default: "unix:control.unit.sock"

  --user=USER          set non-privileged processes to run as specified user
                       default: "nobody"
  --group=GROUP        set non-privileged processes to run as specified group
                       default: user's primary group

  --no-ipv6            disable IPv6 support
  --no-unix-sockets    disable Unix domain sockets support

  --debug              enable debug logging


  python OPTIONS       configure Python module
                       run "./configure python --help" to see available options

  php OPTIONS          configure PHP module
                       run "./configure php --help" to see available options

  go OPTIONS           configure Go module
                       run "./configure go --help" to see available options

  perl OPTIONS         configure Perl module
                       run "./configure perl --help" to see available options

  ruby OPTIONS         configure Ruby module
                       run "./configure ruby --help" to see available options
```